src/rawtoaces_core/CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+cmake_minimum_required(VERSION 3.12)
+include_directories( "${CMAKE_CURRENT_SOURCE_DIR}" )
+
+set( CORE_PUBLIC_HEADER
+    ../../include/rawtoaces/define.h
+    ../../include/rawtoaces/mathOps.h
+    ../../include/rawtoaces/rawtoaces_core.h
+)
+
+add_library( ${RAWTOACES_CORE_LIB} ${DO_SHARED}
+    rawtoaces_core.cpp
+
+    # Make the headers visible in IDEs. This should not affect the builds.
+    ${CORE_PUBLIC_HEADER}
+)
+
+target_link_libraries(
+    ${RAWTOACES_CORE_LIB}
+    PUBLIC
+        Boost::boost
+        Boost::system
+        Boost::filesystem
+        Imath::Imath
+        Imath::ImathConfig
+        Eigen3::Eigen
+)
+
+if ( ${Ceres_VERSION_MAJOR} GREATER 1 )
+    target_link_libraries( ${RAWTOACES_CORE_LIB} PUBLIC Ceres::ceres )
+else ()
+    target_include_directories(${RAWTOACES_CORE_LIB} PUBLIC ${CERES_INCLUDE_DIRS})
+    target_link_libraries(${RAWTOACES_CORE_LIB} PUBLIC ${CERES_LIBRARIES})
+endif ()
+
+if ( LIBRAW_CONFIG_FOUND )
+    target_link_libraries ( ${RAWTOACES_CORE_LIB} PUBLIC libraw::raw )
+else ()
+    target_include_directories(${RAWTOACES_CORE_LIB} PUBLIC ${libraw_INCLUDE_DIR} )
```

Comment:
this one change I've mentioned in prev. comment is on this line:

```c++
get_filename_component(_libraw_parent_inc ${libraw_INCLUDE_DIR} DIRECTORY)
    target_include_directories(${RAWTOACES_CORE_LIB} PUBLIC ${_libraw_parent_inc} ${libraw_INCLUDE_DIR} )
```

I understand it would include prent of where libraw lives as well to search paths list